platform/backend/repository/sqlite/src/main/kotlin/io/hamal/repository/sqlite/record/trigger/Repository.kt

```diff
@@ -118,6 +118,19 @@ class SqliteTriggerRepository(
             if (commandAlreadyApplied(cmdId, triggerId)) {
                 versionOf(triggerId, cmdId) as HookTrigger
             } else {
+
+                list(
+                    TriggerQuery(
+                        hookIds = listOf(cmd.hookId),
+                        funcIds = listOf(cmd.funcId)
+                    )
+                ).firstOrNull()?.let { trigger ->
+                    trigger as HookTrigger
+                    if (trigger.hookMethod == cmd.hookMethod) {
+                        throw IllegalArgumentException("Trigger already exists")
+                    }
+                }
```

Comment:
Good idea - but this should be handled differently:

==> on sqlite level with an partial unique index

see ProjectionUniqueName as reference implementation

Do the same thing for Memory